pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -700,19 +699,24 @@ public ServerReloadControllerJobStatusResponse getReloadJobStatus(
     serverReloadControllerJobStatusResponse.setTotalServersQueried(serverUrls.size());
     serverReloadControllerJobStatusResponse.setTotalServerCallsFailed(serviceResponse._failedResponseCount);
 
+    // Merge all server responses.
+    List<String> pendingSegments = new ArrayList<>();
+    int successCount = 0;
     for (Map.Entry<String, String> streamResponse : serviceResponse._httpResponses.entrySet()) {
       String responseString = streamResponse.getValue();
       try {
         ServerReloadControllerJobStatusResponse response =
             JsonUtils.stringToObject(responseString, ServerReloadControllerJobStatusResponse.class);
-        serverReloadControllerJobStatusResponse.setSuccessCount(
-            serverReloadControllerJobStatusResponse.getSuccessCount() + response.getSuccessCount());
+        successCount += response.getSuccessCount();
+        pendingSegments.addAll(response.getPendingSegments());
```

Comment:
This will throw NPE when server is running the old version

Reply:
hmm. first of all i dont know why this json is decoded into the controller side clas, (where it should really be the server side class) see comment: https://github.com/apache/pinot/pull/8828/files#r1359063941

Reply:
We can change this to `SegmentReloadStatusValue.class` while deserializing.

But, I think since the agreed upon upgrade sequence is controller -> broker -> servers, if new fields are added to `SegmentReloadStatusValue`, in the controller we'd always need to ensure the newer fields are null checked before accessing. Since during upgrade server can be on older version. So moving to `SegmentReloadStatusValue` doesn't change the fact that we'll always need null checks for new fields added to the server's response class?

Reply:
We might also need `JsonIgnoreProperties(ignoreUnknown = true)` annotation on the class it's being deserialised into.